frontend/model/contracts/group.js

```diff
@@ -125,20 +125,37 @@ DefineContract({
       }
     },
     'gi.contracts/group/proposal': {
-      validate: objectOf({
-        proposalType: proposalType,
-        proposalData: object, // data for Vue widgets
-        votingRule: ruleType,
-        expires_date_ms: number // calculate by grabbing proposal expiry from group properties and add to `meta.createdDate`
-      }),
+      validate: data => {
+        objectOf({
+          proposalType: proposalType,
+          proposalData: object, // data for Vue widgets
+          votingRule: ruleType,
+          expires_date_ms: number // calculate by grabbing proposal expiry from group properties and add to `meta.createdDate`
+        })
+        // make sure this isn't a duplicate proposal
+        const ourUsername = sbp('state/vuex/getters').ourUsername
+        const groupID = sbp('state/vuex/state').currentGroupId
+        const groupState = sbp('state/vuex/state')[groupID]
+        for (const hash in groupState.proposals) {
+          const prop = groupState.proposals[hash]
+          if (prop.status === STATUS_OPEN &&
+            prop.data.proposalType === data.proposalType &&
+            prop.meta.username === ourUsername &&
```

Comment:
even if created by a different member, I don't think it should be allowed...

Reply:
I see what you mean, but this is mainly to prevent the kind of bug that I encountered yesterday. It's technically not a bannable offense for someone else to propose the same thing you're proposing. It could be just a mistake